src/components/Company/PaySchedule/PaySchedule.tsx

```diff
@@ -0,0 +1,300 @@
+  createCompoundContext<PayScheduleContextType>('PayScheduleContext')
+export { usePaySchedule }
+
+interface PayScheduleProps extends CommonComponentInterface {
+  companyId: string
+  defaultValues?: PayScheduleDefaultValues
+}
+
+export const PaySchedule = (props: PayScheduleProps & BaseComponentInterface) => {
+  useI18n('Company.PaySchedule')
+  return (
+    <BaseComponent {...props}>
+      <Root {...props}>{props.children}</Root>
+    </BaseComponent>
+  )
+}
+
+const Root = ({ companyId, children, defaultValues }: PayScheduleProps) => {
+  const { baseSubmitHandler, onEvent } = useBase()
+  const [mode, setMode] = useState<MODE>('LIST_PAY_SCHEDULES')
+  const [currentPaySchedule, setCurrentPaySchedule] = useState<
+    (Schemas['Pay-Schedule'] & { version?: string }) | null
+  >(null)
+  const transformedDefaultValues: PayScheduleInputs = useMemo(() => {
+    return {
+      frequency: defaultValues?.frequency ?? 'Every week',
+      anchor_pay_date: defaultValues?.anchor_pay_date
+        ? parseDate(defaultValues.anchor_pay_date)
+        : undefined,
+      anchor_end_of_pay_period: defaultValues?.anchor_end_of_pay_period
+        ? parseDate(defaultValues.anchor_end_of_pay_period)
+        : undefined,
+      day_1: defaultValues?.day_1 ?? undefined,
+      day_2: defaultValues?.day_2 ?? undefined,
+      custom_name: defaultValues?.custom_name ?? '',
+      auto_pilot: defaultValues?.auto_pilot?.toString() ?? '',
+    }
+  }, [defaultValues])
+
+  const [payScheduleDraft, setPayScheduleDraft] = useState<PaySchedulePreviewDraft | null>(null)
+  const {
+    data: payPreviewData,
+    error: paySchedulePreviewError,
+    isLoading,
+  } = useGetPaySchedulePreview(
+    companyId,
+    payScheduleDraft as PaySchedulePreviewDraft, // Casting to non-null because we know it's not null from the enabled prop
+    !!payScheduleDraft,
+  )
+
+  const { data: paySchedules } = useGetAllPaySchedules(companyId)
+
+  const createPayScheduleMutation = useCreatePaySchedule()
+  const updatePayScheduleMutation = useUpdatePaySchedule()
+
+  const formMethods = useForm<PayScheduleInputs, unknown, PayScheduleOutputs>({
+    resolver: valibotResolver(PayScheduleSchema),
+    defaultValues: transformedDefaultValues,
+  })
+
+  const allValues = formMethods.watch()
+
+  useEffect(() => {
+    // Don't update if dates are not set
+    if (!allValues.anchor_pay_date || !allValues.anchor_end_of_pay_period) {
+      return
+    }
+    setPayScheduleDraft({
+      frequency: allValues.frequency as PaySchedulePreviewDraft['frequency'],
+      anchor_pay_date: allValues.anchor_pay_date.toString(),
+      anchor_end_of_pay_period: allValues.anchor_end_of_pay_period.toString(),
+      day_1: allValues.day_1 || undefined,
+      day_2: allValues.day_2 || undefined,
+    })
+  }, [
+    allValues.anchor_end_of_pay_period,
+    allValues.anchor_pay_date,
+    allValues.day_1,
+    allValues.day_2,
+    allValues.frequency,
+    setPayScheduleDraft,
+  ])
```

Comment:
Nit: this seems pretty performance intensive? and if i'm reading this correct this would set the pay schedule draft which would also retrigger the useGetPaySchedulePreview hook? Maybe not a huge deal since it's a limited number of inputs but wondering if there are any perceivable performance issues

Reply:
In terms of perceivable impact it has been minimal. Since we're just changing dates or numbers we're not facing an issue where this needs to be debounced for performance reasons. But this is a good call out i'll triple check we're not facing perf issues.